src/main/main.ts

```diff
@@ -155,3 +157,82 @@
     }
   );
 }
+
+ipcMain.handle('stop-cli', stopCli);
+
+ipcMain.handle('spawn-start-rpc', startRPC);
+
+ipcMain.handle('spawn-balance-check', checkBitcoinBalance);
+
+ipcMain.handle('suspend-current-swap', suspendCurrentSwap);
+
+ipcMain.handle(
+  'spawn-buy-xmr',
+  (_event, provider, redeemAddress, refundAddress) =>
+    buyXmr(redeemAddress, refundAddress, provider)
+);
+
+ipcMain.handle('spawn-cancel-refund', (_event, swapId) =>
+  cancelRefundSwap(swapId)
+);
+
+ipcMain.handle('spawn-resume-swap', (_event, swapId) => resumeSwap(swapId));
+
+ipcMain.handle('spawn-withdraw-btc', (_event, address) =>
+  withdrawAllBitcoin(address)
+);
+
+ipcMain.handle('spawn-list-sellers', (_event, rendezvousPointAddress) =>
+  listSellers(rendezvousPointAddress)
+);
+
+ipcMain.handle('spawn-tor', spawnTor);
+
+ipcMain.handle('stop-tor', stopTor);
+
+ipcMain.handle('get-swap-logs', (_event, swapId) =>
+  getSavedLogsOfSwapId(swapId)
+);
+
+// eslint-disable-next-line import/prefer-default-export
+export function sendSnackbarAlertToRenderer(
+  message: string,
+  variant: string,
+  autoHideDuration: number | null,
+  key: string | null
+) {
+  function send() {
+    logger.debug(
+      { message, variant, autoHideDuration, key },
+      'Attempting to send snackbar alert to renderer'
+    );
+    if (mainWindow) {
+      if (
+        mainWindow.webContents.isDestroyed() ||
+        mainWindow.webContents.isLoading()
+      ) {
+        logger.debug(
+          'Main window is loading, waiting for it to finish before sending snackbar alert'
+        );
+        mainWindow.webContents.once('did-finish-load', () =>
+          setTimeout(send, 5000)
+        );
+      } else {
+        logger.debug(
+          { message, variant, autoHideDuration, key },
+          'Sending snackbar alert to renderer'
+        );
+        mainWindow?.webContents.send(
+          'display-snackbar-alert',
+          message,
+          variant,
+          autoHideDuration,
+          key
+        );
+      }
+    } else {
+      setTimeout(send, 1000);
+    }
+  }
+  send();
+}
```

Comment:
**Suggestion:** Consider using a more robust error handling strategy. Currently, if an error occurs during the execution of the 'sendSnackbarAlertToRenderer' function, the error is simply logged and the execution continues. This might lead to unexpected behavior later in the execution. Consider throwing the error or handling it in a way that prevents further execution in case of a critical failure.
```suggestion
export function sendSnackbarAlertToRenderer(
  message: string,
  variant: string,
  autoHideDuration: number | null,
  key: string | null
) {
  function send() {
    logger.debug(
      { message, variant, autoHideDuration, key },
      'Attempting to send snackbar alert to renderer'
    );
    if (mainWindow) {
      if (
        mainWindow.webContents.isDestroyed() ||
        mainWindow.webContents.isLoading()
      ) {
        logger.debug(
          'Main window is loading, waiting for it to finish before sending snackbar alert'
        );
        mainWindow.webContents.once('did-finish-load', () =>
          setTimeout(send, 5000)
        );
      } else {
        logger.debug(
          { message, variant, autoHideDuration, key },
          'Sending snackbar alert to renderer'
        );
        try {
          mainWindow?.webContents.send(
            'display-snackbar-alert',
            message,
            variant,
            autoHideDuration,
            key
          );
        } catch (e) {
          logger.error(
            { error: e.toString() },
            'Failed to send snackbar alert to renderer'
          );
          throw e;
        }
      }
    } else {
      setTimeout(send, 1000);
    }
  }
  send();
}
```